packages/core/src/baseclient.ts

```diff
@@ -110,24 +103,15 @@ export abstract class BaseClient<B extends Backend, O extends Options> implement
    */
   public captureMessage(message: string, level?: Severity, hint?: EventHint, scope?: Scope): string | undefined {
     let eventId: string | undefined = hint && hint.event_id;
-
     this._processing = true;
 
     const promisedEvent = isPrimitive(message)
       ? this._getBackend().eventFromMessage(`${message}`, level, hint)
       : this._getBackend().eventFromException(message, hint);
 
-    promisedEvent
-      .then(event => this._processEvent(event, hint, scope))
-      .then(finalEvent => {
-        // We need to check for finalEvent in case beforeSend returned null
-        eventId = finalEvent && finalEvent.event_id;
-        this._processing = false;
-      })
-      .then(null, reason => {
-        logger.error(reason);
-        this._processing = false;
-      });
+    promisedEvent.then(event => {
+      eventId = this.captureEvent(event, hint, scope);
+    });
```

Comment:
@kamilogorek No idea why we didn't do it like this in the first place.

Reply:
Good catch

Reply:
Same here. Whole method can be written as:

```js
const promisedEvent = // ...;

return promisedEvent.then(event => this.captureEvent(event, hint, scope);
```